ios/CameraView+RecordVideo.swift

```diff
@@ -189,7 +192,43 @@ extension CameraView: AVCaptureVideoDataOutputSampleBufferDelegate, AVCaptureAud
     }
   }
 
+  public final func dataOutputSynchronizer(_: AVCaptureDataOutputSynchronizer, didOutput synchronizedDataCollection: AVCaptureSynchronizedDataCollection) {
+    // Read all outputs
+    guard let syncedVideoData: AVCaptureSynchronizedSampleBufferData =
+      synchronizedDataCollection.synchronizedData(for: videoOutput!) as? AVCaptureSynchronizedSampleBufferData else {
+      // only work on synced pairs
+      ReactLogger.log(level: .info, message: "Video data out of sync")
+      return
+    }
+    guard let syncedDepthData: AVCaptureSynchronizedDepthData =
+      synchronizedDataCollection.synchronizedData(for: depthOutput!) as? AVCaptureSynchronizedDepthData else {
+      ReactLogger.log(level: .info, message: "Depth data out of sync")
+      return
+    }
+
+    if syncedDepthData.depthDataWasDropped || syncedVideoData.sampleBufferWasDropped {
+      return
+    }
+
+    let depthData = syncedDepthData.depthData
+    let depthPixelBuffer = depthData.depthDataMap
+    let sampleBuffer = syncedVideoData.sampleBuffer
+
+    if let frameProcessor = frameProcessorCallback {
+      if !isRunningFrameProcessor {
+        // we're not in the middle of executing the Frame Processor, so prepare for next call.
+        CameraQueues.frameProcessorQueue.async {
+          self.isRunningFrameProcessor = true
+          let frame = Frame(bufferAndDepth: sampleBuffer, depth: depthPixelBuffer, orientation: self.bufferOrientation)
+          frameProcessor(frame)
+          self.isRunningFrameProcessor = false
+        }
+      }
+    }
```

Comment:
We could probably create a method that wraps all of this logic so we only call the call constructor once, because this code is mostly duplicated.

E.g. something like:

```swift
if isFrameProcessorReady {
  callFrameProcessor(withFrame: Frame(..))
}
```

But I'll do that refactor later. Leave it as is now

Reply:
Yeah that would be brill - will leave for now :)